# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: gfx/angle/checkout/src/libANGLE/renderer/TextureImpl.h
# Commit: 3a91885d47e0
# Full Hash: 3a91885d47e0ef7fba9749562d52300eea67c651
# Author: Miko Mynttinen <mikokm@gmail.com>
# Date: 2019-11-11 21:52:52
# Regressor Bug: 1578576
# File Overlap Count: 4
# Description:
#   Bug 1578576 - Part 2: Update to ANGLE 3865 r=jgilbert
#   
#   Depends on D44561
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D44579
# ==============================================================================

diff -r 77cee77c0307 -r 3a91885d47e0 gfx/angle/checkout/src/libANGLE/renderer/TextureImpl.h
--- a/gfx/angle/checkout/src/libANGLE/renderer/TextureImpl.h	Fri Nov 08 16:41:42 2019 +0000
+++ b/gfx/angle/checkout/src/libANGLE/renderer/TextureImpl.h	Fri Nov 08 17:06:00 2019 +0000
@@ -32,6 +32,7 @@
 struct Offset;
 struct Rectangle;
 class Framebuffer;
+class MemoryObject;
 struct PixelUnpackState;
 class TextureState;
 }  // namespace gl
@@ -147,6 +148,21 @@
                                                 const gl::Extents &size,
                                                 bool fixedSampleLocations) = 0;
 
+    virtual angle::Result setStorageExternalMemory(const gl::Context *context,
+                                                   gl::TextureType type,
+                                                   size_t levels,
+                                                   GLenum internalFormat,
+                                                   const gl::Extents &size,
+                                                   gl::MemoryObject *memoryObject,
+                                                   GLuint64 offset) = 0;
+
+    virtual angle::Result setImageExternal(const gl::Context *context,
+                                           const gl::ImageIndex &index,
+                                           GLenum internalFormat,
+                                           const gl::Extents &size,
+                                           GLenum format,
+                                           GLenum type);
+
     virtual angle::Result setEGLImageTarget(const gl::Context *context,
                                             gl::TextureType type,
                                             egl::Image *image) = 0;
@@ -167,6 +183,8 @@
     virtual GLint getMemorySize() const;
     virtual GLint getLevelMemorySize(gl::TextureTarget target, GLint level);
 
+    virtual GLint getNativeID() const;
+
     virtual angle::Result syncState(const gl::Context *context,
                                     const gl::Texture::DirtyBits &dirtyBits) = 0;
 